mobile-widgets/qmlmanager.cpp

```diff
@@ -927,6 +927,7 @@ bool QMLManager::checkDuration(DiveObjectHelper *myDive, struct dive *d, QString
 			free(d->dc.sample);
 			d->dc.sample = 0;
 			d->dc.samples = 0;
+			d->dc.alloc_samples = 0;
```

Comment:
It is kind of sad that the UI-code directly manipulates internal data structures.

Therefore, in the short-run, I would suggest creating a `free_samples()` function, which does the above and is called here and in `planner.c`. But I think this is beyond the scope of this PR.

In the long run, I think we should reimplement `std::vector<>` in C [my personal choice would be simply using `std::vector<>` in the backend (while still providing a C-only interface). But no worries - I'm not going to try to convince anybody.]

Reply:
I agree that making  a free_samples function is the minimum we should do, but I wanted to keep this PR as small as possible (as you also write out of scope). If/when this PR is merged I am happy to make the function.

I agree with you on the vector considerations.

Reply:
C++ code compiles slower and runs usually an order of magnitude slower. Which is why we generally don't jump on that bandwagon for things that can just as easily done in C.